R/github-actions.R

```diff
@@ -152,15 +118,97 @@ use_github_action <- function(name,
     ui_todo("Learn more at <{readme}>.")
   }
 
+  badge <- badge %||% is_check_action(url)
+  if (badge) {
+    use_github_actions_badge(save_as)
+  }
+
   invisible(new)
 }
 
-#' @section `use_github_action_check_release()`:
-#' This entry-level, bare-minimum workflow installs the latest release of R (on
-#' a current distribution of Linux) and runs `R CMD check` via the
-#' [rcmdcheck](https://github.com/r-lib/rcmdcheck) package.
+choose_gha_workflow <- function(error_call = caller_env()) {
+  if (!is_interactive()) {
+    cli::cli_abort(
+      "{.arg name} is absent and must be supplied",
+      call = error_call
+    )
+  }
+
+  prompt <- cli::format_inline(
+    "Which action do you want to add? (0 to exit)\n",
+    "(See {.url https://github.com/r-lib/actions/tree/v2/examples} for other options)"
+  )
+  # Any changes here also need to be reflected in documentation
+  workflows <- c(
+    "check-standard" = "Run R CMD check in four environments",
+    "test-coverage" = "Compute test coverage and report to https://about.codecov.io",
+    "lint" = "Run lintr::lint_package() and report problems",
+    "pr-commands" = "Add /document and /style commands for pull requests"
+  )
+  options <- paste0(cli::style_bold(names(workflows)), ": ", workflows)
+
+  choice <- utils::menu(
+    title = prompt,
+    choices = options
+  )
+  if (choice == 0) {
+    cli::cli_abort("Selection terminated", call = error_call)
+  }
+
+  names(workflows)[choice]
+}
+
+is_check_action <- function(url) {
+  grepl("^check-", path_file(url))
+}
+
+#' Generates a GitHub Actions badge
+#'
+#' Generates a GitHub Actions badge and that's all. This exists primarily for
+#' internal use.
+#'
+#' @keywords internal
+#' @param name Name of the workflow's YAML configuration file (with or without
+#'   extension), e.g. `"R-CMD-check"`, `"R-CMD-check.yaml"`.
+#' @inheritParams use_github_action
+#' @export
+use_github_actions_badge <- function(name = "R-CMD-check.yaml",
+                                     repo_spec = NULL) {
+  if (path_ext(name) == "") {
+    name <- path_ext_set(name, "yaml")
+  }
+  repo_spec <- repo_spec %||% target_repo_spec()
+  enc_name <- utils::URLencode(name)
+  img <- glue("https://github.com/{repo_spec}/actions/workflows/{enc_name}/badge.svg")
+  url <- glue("https://github.com/{repo_spec}/actions/workflows/{enc_name}")
+
+  use_badge(path_ext_remove(name), url, img)
+}
+
+# deprecated --------------------------------------------------------------
+
+#' Use specific Github actions
+#'
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' `use_github_actions()` is deprecated because it was just an alias
+#' for [use_github_action_check_release()]. The other variants are deprecated
+#' because [use_github_action()] can  now suggest specific workflows to use.
```

Comment:
```suggestion
#' Use specific GitHub Actions workflows
#'
#' @description
#' `r lifecycle::badge("deprecated")`
#'
#' `use_github_actions()` is deprecated because it was just an alias
#' for [use_github_action_check_release()]. The other variants are deprecated
#' because [use_github_action()] can now suggest specific workflows to use.
```